autoscale/src/test/java/com/sequenceiq/periscope/endpointtests/DistroXAutoScaleScalingActivityV1EndpointTest.java

```diff
@@ -0,0 +1,485 @@
+                .getScalingActivitiesInDurationByClusterCrn(TEST_CLUSTER_CRN_NULL, 60, 0, 10));
+    }
+
+    @Test
+    public void testGetScalingActivityUsingActivityCrn() {
+        when(threadLocalService.getCloudbreakTenant()).thenReturn(TEST_TENANT);
+        DistroXAutoscaleScalingActivityResponse distroXAutoscaleScalingActivityResponse = distroXAutoScaleScalingActivityV1Endpoint
+                .getScalingActivityUsingActivityCrn(TEST_ACTIVITY_CRN);
+        assertEquals(TEST_ACTIVITY_CRN, distroXAutoscaleScalingActivityResponse.getActivityCrn());
+    }
+
+    @Test
+    public void testGetScalingActivityUsingActivityCrnWithCrnNotFound() {
+        assertThrows(NotFoundException.class, () -> distroXAutoScaleScalingActivityV1Endpoint
+                .getScalingActivityUsingActivityCrn(TEST_ACTIVITY_CRN_9));
+    }
+
+    @Test
+    public void testGetScalingActivityUsingActivityCrnWithNullCrn() {
+        assertThrows(NotFoundException.class, () -> distroXAutoScaleScalingActivityV1Endpoint
+                .getScalingActivityUsingActivityCrn(TEST_ACTIVITY_CRN_NULL));
+    }
+
+    @Test
+    public void testGetFailedScalingActivitiesInGivenDurationByClusterName() {
+        List<String> distroXAutoscaleScalingActivityResponse = distroXAutoScaleScalingActivityV1Endpoint
+                .getFailedScalingActivitiesInGivenDurationByClusterName(TEST_CLUSTER_NAME, 60, 0, 3)
+                .stream().map(DistroXAutoscaleScalingActivityResponse::getActivityCrn).collect(Collectors.toList());
+        assertEquals(distroXAutoscaleScalingActivityResponse, List.of(TEST_ACTIVITY_CRN_6, TEST_ACTIVITY_CRN_4, TEST_ACTIVITY_CRN_3));
+    }
+
+    @Test
+    public void testGetFailedScalingActivitiesInGivenDurationByClusterNameWithNoFailedScalingActivity() {
+        List<String> distroXAutoscaleScalingActivityResponse = distroXAutoScaleScalingActivityV1Endpoint
+                .getFailedScalingActivitiesInGivenDurationByClusterName(TEST_CLUSTER_NAME_2, 60, 0, 3)
+                .stream().map(DistroXAutoscaleScalingActivityResponse::getActivityCrn).collect(Collectors.toList());
+        assertEquals(distroXAutoscaleScalingActivityResponse, List.of());
+    }
+
+    @Test
+    public void testGetFailedScalingActivitiesInGivenDurationByClusterNameWithNoScalingActivityStored() {
+        List<String> distroXAutoscaleScalingActivityResponse = distroXAutoScaleScalingActivityV1Endpoint
+                .getFailedScalingActivitiesInGivenDurationByClusterName(TEST_CLUSTER_NAME_1, 60, 0, 3)
+                .stream().map(DistroXAutoscaleScalingActivityResponse::getActivityCrn).collect(Collectors.toList());
+        assertEquals(distroXAutoscaleScalingActivityResponse, List.of());
+    }
+
+    @Test
+    public void testGetFailedScalingActivitiesInGivenDurationByClusterNameWithNullName() {
+        assertThrows(InternalServerErrorException.class, () -> distroXAutoScaleScalingActivityV1Endpoint
```

Comment:
Is `InternalServerError` the right exception that should be thrown in this case? `BadRequestException` makes more sense. `InternalServerError` will show up later on as Datadog alerts, since it has a 500 response code